traitlets/utils/bunch.py

```diff
@@ -16,3 +16,9 @@ def __getattr__(self, key):
     
     def __setattr__(self, key, value):
         self.__setitem__(key, value)
+    
+    def __dir__(self):
+        names = super(Bunch, self).__dir__()
+        names.extend(self.keys())
+        return names
```

Comment:
Turns out, you can't on Python 2. I did something a bit worse so that it works on both reasonable Python and Python 2.